### PR TITLE
Create button to nullify date_time fields

### LINKED ISSUE #2770: Can't nullify the date/date_time field
### Steps to Reproduce
1. Go to a resource with a date/date_time field
2. set a value and save
3. go back to the edit page
4. try to make it null or empty

### Expected behavior & Actual behavior

You should be able to empty that field.

The button can look something like this but prettier with a tooltip.

CleanShot 2024-05-30 at 16 36 05@2x

### PR DESCRIPTION
**Pull request by binarygit**
# Description
This PR adds a button to nullify date_time fields.

Fixes #2770

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/avo-hq/avodocs)
- [x] I have added tests that prove my fix is effective or that my feature works

## Screenshots & recording
This is how it looks right now:
crop
image
reset

Manual reviewer: please leave a comment with output from the test if that's the case.

### CODE REVIEW

**Comment by binarygit** [comment]
I haven't found a good way to align the button horizontally to the input field. I poked around the css but no luck.

My approach is to add the button inside the `app/components/avo/field_wrapper_component.html.erb` and then render it conditionally. However I don't think it's a good idea which is why I wanted to ask you guys if you see/know an easier way to have the button render horizontal to the input field.

**Comment by binarygit** [comment]
Tests pass locally. Not sure why CI's fails :shrug:

**Comment by adrianthedev** [comment]
What do you think about this type of button @binarygit?

CleanShot 2024-07-08 at 14 32 45

**Comment by adrianthedev** [comment]
The linter is [failing](https://github.com/avo-hq/avo/actions/runs/9840142380/job/27163697899?pr=2951#step:12:54), and with my suggestion as well. Let's lowercase the `reset` translation, please.

**Comment by binarygit** [comment]
It's done

**Review by adrianthedev — CHANGES_REQUESTED** [review]
It's almost done to merge.
We need to apply the changes to the other field as well.

File: app/components/avo/fields/time_field/edit_component.html.erb [hunk 1]
```diff
@@ -36,5 +36,11 @@
       placeholder: @field.placeholder,
       style: @field.get_html(:style, view: view, element: :input)
     %>
+    <%= content_tag :button, id: :reset, type: :button,
+                    title: "Reset Time",
+                    data: {action: "click->date-field#clear",
+                           tippy: :tooltip } do %>
+      <%= render Avo::Fields::Common::BooleanCheckComponent.new %>
```
[the remark this answers is not in the document]
Comment (binarygit) [reply app/components/avo/fields/time_field/edit_component.html.erb]: This comment completely flew under my radar! I have just made the requested change.

**Pull request merged**